IdentityCore/src/util/MSIDKeychainUtil.m

```diff
@@ -29,51 +29,81 @@ @implementation MSIDKeychainUtil
-
-+ (NSString *)retrieveTeamIDFromKeychain
++ (NSString *)appDefaultAccessGroup
 {
-    NSDictionary *query = @{ (id)kSecClass : (id)kSecClassGenericPassword,
-                             (id)kSecAttrAccount : @"teamIDHint",
-                             (id)kSecAttrService : @"",
-                             (id)kSecReturnAttributes : @YES };
-    CFDictionaryRef result = nil;
+    static dispatch_once_t once;
+    static NSString *appDefaultAccessGroup = nil;
     
-    OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query, (CFTypeRef *)&result);
+    dispatch_once(&once, ^{
+        NSDictionary *query = @{ (id)kSecClass : (id)kSecClassGenericPassword,
+                                 (id)kSecAttrAccount : @"teamIDHint",
+                                 (id)kSecAttrService : @"",
+                                 (id)kSecReturnAttributes : @YES };
+        CFDictionaryRef result = nil;
+        
+        OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query, (CFTypeRef *)&result);
+        
+        if (status == errSecItemNotFound)
+        {
+            NSMutableDictionary* addQuery = [query mutableCopy];
+            [addQuery setObject:(id)kSecAttrAccessibleAlways forKey:(id)kSecAttrAccessible];
+            status = SecItemAdd((__bridge CFDictionaryRef)addQuery, (CFTypeRef *)&result);
+        }
+        
+        if (status == errSecSuccess)
+        {
+            appDefaultAccessGroup = [(__bridge NSDictionary *)result objectForKey:(__bridge id)(kSecAttrAccessGroup)];
+            MSID_LOG_INFO(nil, @"Defaul app's acces group: \"%@\".", _PII_NULLIFY(appDefaultAccessGroup));
+            MSID_LOG_INFO_PII(nil, @"Defaul app's acces group: \"%@\".", appDefaultAccessGroup);
+        }
+        else
+        {
+            MSID_LOG_ERROR(nil, @"fetching kSecAttrAccessGroup, status: %d", status);
+        }
+        
+        CFRelease(result);
+    });
     
-    if (status == errSecItemNotFound)
+    return appDefaultAccessGroup;
+}
+
++ (NSString *)accessGroup:(NSString *)group
+{
+    if (!group)
     {
-        NSMutableDictionary* addQuery = [query mutableCopy];
-        [addQuery setObject:(id)kSecAttrAccessibleAlways forKey:(id)kSecAttrAccessible];
-        status = SecItemAdd((__bridge CFDictionaryRef)addQuery, (CFTypeRef *)&result);
+        return nil;
     }
     
-    if (status != errSecSuccess)
+    if (!MSIDKeychainUtil.teamId)
     {
-        MSID_LOG_ERROR(nil, @"fetching team ID, status: %d", status);
         return nil;
     }
     
-    NSString *accessGroup = [(__bridge NSDictionary *)result objectForKey:(__bridge id)(kSecAttrAccessGroup)];
-    NSArray *components = [accessGroup componentsSeparatedByString:@"."];
-    NSString *bundleSeedID = [components firstObject];
-    
-    CFRelease(result);
+#if TARGET_OS_SIMULATOR
+    // In simulator team id can be "FAKETEAMID" (for example in UT without host app).
+    if ([MSIDKeychainUtil.teamId isEqualToString:@"FAKETEAMID"])
```

Comment:
where is this coming from (FAKETEAMID)?

Reply:
If we run the code in UT without "Host App", the team id will be "FAKETEAMID". We get it when we are getting app's default access group (see [MSIDKeychainUtil appDefaultAccessGroup];)